library/tasks/HISAT2.wdl

```diff
@@ -0,0 +1,299 @@
+task HISAT2PairedEnd {
+  meta {
+    description: "JISHUUUUUUUUUUUUUUUU"
+  }
+
+  File hisat2_ref
+  File fq1
+  File fq2
+  String ref_name
+  String output_name
+  String sample_name
+
+  # runtime optional arguments
```

Comment:
The idea for this process is that there are explicit optional arguments for the task that then get converted to non optional values though some `select_first([optional_value, default_value])` expression.  Things I like about this is that its very explicit as to what is expected for the user to put an input to if they wish to change a runtime value.  I don't like how much boilerplate code this adds to every task though.

Reply:
It seems that the only additional code in this task is this block:
```wdl
 String? opt_docker
 Int? opt_memory_gb
 Int? opt_cpu
 Int? opt_disk
 Int? opt_preemptible
```
Is that right? If that's the only additional boilerplate over the following version, then this option seems clearest to the user in my opinion. That said, I haven't used `parameter_meta` before, so maybe I'm biased.

Reply:
yup those 5 inputs are the "extra boilerplate". I wasn't sure if 5 lines per task were too much but good to hear that you think its acceptable!